azure/durable_functions/orchestrator.py

```diff
@@ -49,57 +49,66 @@ def handle(self, context: DurableOrchestrationContext):
         suspended = False
 
         fn_output = self.fn(self.durable_context)
+
         # If `fn_output` is not an Iterator, then the orchestrator
         # function does not make use of its context parameter. If so,
         # `fn_output` is the return value instead of a generator
-        if isinstance(fn_output, Iterator):
-            self.generator = fn_output
-
-        else:
+        if not isinstance(fn_output, Iterator):
```

Comment:
I inverted some if-else logic, which frees us from having to check for `continue_as_new` in so many places